### PR TITLE
Wrong typ of insert by adding a new MessageIcon fixes #5033

### LINKED ISSUE #5033: Insert a new message icon, create a db syntax error
#### Description
When you insert a new message icon a db syntax error is raised up,
when postgres is used.

grafik

### Steps to reproduce
1. Add a new message icon
2. 

### Environment (complete as necessary)
- Version/Git revision: newest
- Database Type: Postgres
- Database Version: 10
- PHP Version: 7.1

### Additional information/references

get fixed by https://github.com/SimpleMachines/SMF2.1/pull/5031

### PR DESCRIPTION
**Pull request by albertlast**
Replace methode can only be used,
when pk field is mention as field.

Sanity Check added

Signed-off-by: albertlast albertlast@hotmail.de

### CODE REVIEW

File: Sources/Subs-Db-postgresql.php [hunk 1]
```diff
@@ -717,6 +717,11 @@ function smf_db_insert($method = 'replace', $table, $columns, $data, $keys, $ret
 	// Replace the prefix holder with the actual prefix.
 	$table = str_replace('{db_prefix}', $db_prefix, $table);
 
+	// Sanity check for replace is key part of the columns array
+	if ($method == 'replace' && count(array_intersect_key($columns, array_flip($keys))) !== count($keys))
+		smf_db_error_backtrace('Primary Key field missing in insert call',
+				'Change the methode of db insert to insert or add the pk field to the columns array', E_USER_ERROR, __FILE__, __LINE__);
```
Comment (Sesquipedalian) [inline Sources/Subs-Db-postgresql.php]: These strings should be loaded using the `$txt` variable instead of hard-coding the English text.

**Review by Sesquipedalian — CHANGES_REQUESTED** [review]
Please use `$txt` to load strings for the backtrace message.

File: Sources/Subs-Db-postgresql.php (hunk 1 above)
Comment (albertlast) [reply Sources/Subs-Db-postgresql.php]: When you look in the other backtrace you notice that every where is written in hardcode and
specialy database stuff is not the rightplace for this.

When you wanted feel free to do this but than please for the complet backtrace stuff.

**Comment by albertlast** [comment]
Any other issue @Sesquipedalian or is this pr now fine?

Your $txt request is in my eyes a own pr because the language file didn't exists,
create the complet env would overextend the scope of this pr.

**Review by Sesquipedalian — APPROVED** [review]

File: Sources/Subs-Db-postgresql.php (hunk 1 above)
Comment (Sesquipedalian) [reply Sources/Subs-Db-postgresql.php]: Fair enough. Then please just change "methode" to "method".

**Comment by Sesquipedalian** [comment]
Yes, that makes sense. Please just correct the spelling mistake in the message, and then I will merge.

**Comment by albertlast** [comment]
Fixed spelling

**Pull request merged**